storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -577,288 +409,265 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
     }
+    return retVal;
+  }
 
-    for (OfferID id : toLaunch.keySet()) {
-      List<LaunchTask> tasks = toLaunch.get(id);
-      List<TaskInfo> launchList = new ArrayList<>();
+  private Resource createMesosRangeResource(ResourceType resourceType, RangeResourceEntry rangeResourceEntry) {
+    Ranges.Builder rangesBuilder = Ranges.newBuilder();
+    Range rangeBuilder = Range.newBuilder()
+                              .setBegin(rangeResourceEntry.getBegin())
+                              .setEnd(rangeResourceEntry.getEnd()).build();
+    rangesBuilder.addRange(rangeBuilder);
+
+    Resource.Builder resourceBuilder = Resource.newBuilder()
+                                               .setName(resourceType.toString())
+                                               .setType(Protos.Value.Type.RANGES)
+                                               .setRanges(rangesBuilder.build());
+    if (rangeResourceEntry.getReservationType() != null && rangeResourceEntry.getReservationType().equals(ReservationType.STATICALLY_RESERVED)) {
+      resourceBuilder.setRole(MesosCommon.getRole(mesosStormConf));
+    }
+    return resourceBuilder.build();
+  }
 
-      LOG.info("Launching tasks for offerId: {} : {}", id.getValue(), launchTaskListToString(tasks));
-      for (LaunchTask t : tasks) {
-        launchList.add(t.getTask());
-        taskIDtoOfferMap.put(t.getTask().getTaskId(), t.getOffer());
-      }
+  String getFullConfigUri() {
+    try {
+      return new URL(_configUrl.toURL(),
+                 _configUrl.getPath() + "/storm.yaml").toString();
+    } catch (MalformedURLException e) {
+      throw new RuntimeException(e);
+    }
+  }
 
-      List<OfferID> launchOffer = new ArrayList<>();
-      launchOffer.add(id);
-      _driver.launchTasks(launchOffer, launchList);
-      _offers.remove(id);
+  private ExecutorInfo.Builder getExecutorInfoBuilder(TopologyDetails details, String executorDataStr,
+                                                      String executorName,
+                                                      List<Resource> executorCpuResources,
+                                                      List<Resource> executorMemResources,
+                                                      List<Resource> executorPortsResources,
+                                                      String extraConfig) {
+    String configUri;
+
+    configUri = getFullConfigUri();
+
+    ExecutorInfo.Builder executorInfoBuilder = ExecutorInfo.newBuilder();
+
+    executorInfoBuilder
+      .setName(executorName)
+      .setExecutorId(ExecutorID.newBuilder().setValue(details.getId()))
+      .setData(ByteString.copyFromUtf8(executorDataStr))
+      .addAllResources(executorCpuResources)
+      .addAllResources(executorMemResources)
+      .addAllResources(executorPortsResources);
+
+    ICommandLineShim commandLineShim = CommandLineShimFactory.makeCommandLineShim(_container.isPresent(), extraConfig);
+    if (_container.isPresent()) {
```

Comment:
maybe slap in a comment about the forking here being based on choice of docker or mesos containerizer?